pvlib/iotools/nasa_power.py

```diff
@@ -0,0 +1,153 @@
+        interest. The default is ``'re'``. Note that in many cases this choice
+        might affect the units of the parameter.
+
+        * ``'re'``: renewable energy
+        * ``'sb'``: sustainable buildings
+        * ``'ag'``: agroclimatology
+
+    elevation: float, optional
+        The custom site elevation in meters to produce the corrected
+        atmospheric pressure adjusted for elevation.
+    wind_height: float, optional
+        The custom wind height in meters to produce the wind speed adjusted
+        for height. Has to be between 10 and 300 m; see [4]_.
+    wind_surface: str, optional
+        The definable surface type to adjust the wind speed. For a list of the
+        surface types see [4]_. If you provide a wind surface alias please
+        include a site elevation with the request.
+    map_variables: bool, optional
+        When true, renames columns of the Dataframe to pvlib variable names
+        where applicable. See variable :const:`VARIABLE_MAP`.
+        The default is `True`.
+
+    Raises
+    ------
+    requests.HTTPError
+        Raises an error when an incorrect request is made.
+
+    Returns
+    -------
+    data : pd.DataFrame
+        Time series data. The index corresponds to the start (left) of the
+        interval.
+    meta : dict
+        Metadata.
+
+    References
+    ----------
+    .. [1] `NASA Prediction Of Worldwide Energy Resources (POWER)
+       <https://power.larc.nasa.gov/>`_
+    .. [2] `NASA POWER API
+       <https://power.larc.nasa.gov/api/pages/>`_
+    .. [3] `NASA POWER API parameters
+       <https://power.larc.nasa.gov/parameters/>`_
+    .. [4] `NASA POWER corrected wind speed parameters
+       <https://power.larc.nasa.gov/docs/methodology/meteorology/wind/>`_
+    """
+    start = pd.Timestamp(start)
+    end = pd.Timestamp(end)
+
+    # allow the use of pvlib parameter names
+    parameter_dict = {v: k for k, v in VARIABLE_MAP.items()}
+    parameters = [parameter_dict.get(p, p) for p in parameters]
+
+    params = {
+        'latitude': latitude,
+        'longitude': longitude,
+        'start': start.strftime('%Y%m%d'),
+        'end': end.strftime('%Y%m%d'),
+        'community': community,
+        'parameters': ','.join(parameters),  # make parameters in a string
+        'format': 'json',
+        'user': None,
+        'header': True,
+        'time-standard': 'utc',
+        'site-elevation': elevation,
+        'wind-elevation': wind_height,
+        'wind-surface': wind_surface,
+    }
+
+    response = requests.get(url, params=params)
+    if not response.ok:
+        # response.raise_for_status() does not give a useful error message
+        raise requests.HTTPError(response.json())
+
+    # Parse the data to dataframe
+    data = response.json()
+    hourly_data = data['properties']['parameter']
+    df = pd.DataFrame(hourly_data)
+    df.index = pd.to_datetime(df.index, format='%Y%m%d%H').tz_localize('UTC')
+    df = df.replace(-999, np.nan)
```

Comment:
perhaps worth using `meta['fill_value']` instead of hardcoding -999, in case they change it someday?